frontend/src/components/student/AllActivities/CombatTask/CombatTask.js

```diff
@@ -81,69 +78,131 @@ export default function CombatTask() {
+          </HeaderRow>
+          <VerticalSpacer />
+          <Row
+            className='p-2 rounded mx-2 overflow-auto'
+            style={{ backgroundColor: 'var(--dark-blue)', height: '25vh' }}
+          >
+            <ActivityDetails />
+          </Row>
+          <VerticalSpacer />
+          <Row className='p-2 rounded mx-2' style={{ backgroundColor: 'var(--dark-blue)', height: '50vh' }}>
+            <Col
+              md={task.answer || hasSentNow ? MD_WHEN_TASK_SENT : MD_WHEN_TASK_NOT_SENT}
+              className={'h-100 overflow-auto'}
+            >
+              <h4>Odpowiedź:</h4>
+              {isRevieved() ? (
+                <Col>
+                  <h4>Twoja odpowiedź</h4>
+                  <p>{task.answer}</p>
+                </Col>
+              ) : (
                 <>
-                  <SmallDivider />
-                  <p>
-                    Zdobyte punkty: <strong>{task.points}</strong>
-                  </p>
-                  <p>Uwagi od prowadzącego:</p>
-                  <p>{task.remarks ?? 'Brak uwag'}</p>
-                  {task.feedbackFile && <FeedbackFileService feedbackFile={task.feedbackFile} />}
-                </>
-              )}
-              <SmallDivider />
-              {task.points == null && (
-                <RemarksCol>
-                  <h4>Odpowiedź:</h4>
+                  {task.answer ? (
+                    <Col>
+                      <h5>Twoja obecna odpowiedź</h5>
+                      <p>{task.answer}</p>
+                    </Col>
+                  ) : null}
                   <RemarksTextArea
                     ref={textAreaRef}
+                    disabled={isRevieved()}
                     onChange={(e) => {
                       handleAnswerChange(e)
                     }}
                   />
-                </RemarksCol>
+                </>
               )}
-              <FileService
-                task={task}
-                setFile={setFileBlob}
-                setFileName={setFileName}
-                setIsFetching={setIsFetching}
-                isFetching={isFetching}
-                isRevieved={task.points != null}
-              />
-            </div>
-            <SendTaskButton disabled={task.points !== null} onClick={sendAnswer}>
-              {isFetching ? (
-                <Spinner animation={'border'} />
-              ) : task.points == null ? (
-                <span>Wyślij</span>
+              <Col className={'text-center overflow-auto'}>
+                <FileService
+                  task={task}
+                  setFile={setFileBlob}
+                  setFileName={setFileName}
+                  setIsFetching={setIsFetching}
+                  isFetching={isFetching}
+                  isRevieved={isRevieved()}
+                />
+              </Col>
+              <Col className={'w-100 text-center'}>
+                <SendTaskButton disabled={task.points !== null} onClick={sendAnswer}>
```

Comment:
Czy 
```js 
disabled={task.points}
``` 
nie da tego samego efektu?

Reply:
Sam się na to naciąłem xd, kiedy ktoś przyzna 0 pkt to jest falsy według JS-a, więc ten check jest konieczny niestety

Reply:
w takim razie wystarczy !=
Dla null nigdy nie trzeba potrójnego =